lib/util.js

```diff
@@ -804,3 +805,95 @@ exports.validatePath = function (dir) {
     return false;
   }
 };
+
+exports.getEnvVar = function (variable) {
+  return process.env[variable.toLowerCase()] || process.env[variable.toUpperCase()];
+};
+
+exports.validateProxy = function (proxy) {
+  const { host, port, noProxy, user, password } = proxy;
+  // check for missing proxyHost
+  Errors.checkArgumentExists(this.exists(host),
+    ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_HOST);
+
+  // check for invalid proxyHost
+  Errors.checkArgumentValid(this.isString(host),
+    ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_HOST);
+
+  // check for missing proxyPort
+  Errors.checkArgumentExists(this.exists(port),
+    ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_PORT);
+
+  // check for invalid proxyPort
+  Errors.checkArgumentValid(this.isNumber(port),
+    ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_PORT);
+
+  if (this.exists(noProxy)) {
+    // check for invalid noProxy
+    Errors.checkArgumentValid(this.isString(noProxy),
+      ErrorCodes.ERR_CONN_CREATE_INVALID_NO_PROXY);
+  }
+
+  if (this.exists(user) || this.exists(password)) {
+    // check for missing proxyUser
+    Errors.checkArgumentExists(this.exists(user),
+      ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_USER);
+
+    // check for invalid proxyUser
+    Errors.checkArgumentValid(this.isString(user),
+      ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_USER);
+
+    // check for missing proxyPassword
+    Errors.checkArgumentExists(this.exists(password),
+      ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_PASS);
+
+    // check for invalid proxyPassword
+    Errors.checkArgumentValid(this.isString(password),
+      ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_PASS);
+
+  } else {
+    delete proxy.user;
+    delete proxy.password;
+  }
+};
+
+exports.validateEmptyString = function (value) {
+  return value !== '' ? value : undefined;
+};
+
+exports.getProxyEnv = function (isHttps) {
+  let proxyFromEnv;
+  if (!isHttps) {
+    proxyFromEnv = this.getEnvVar('http_proxy');
+    Logger.getInstance().info('Util.getProxyEnv: Using HTTP_PROXY from the environment variable');
+  } else {
+    proxyFromEnv = this.getEnvVar('https_proxy');
+    Logger.getInstance().info('Util.getProxyEnv: Using HTTPS_PROXY from the environment variable');
+  }
+  
+  if (!proxyFromEnv) {
+    return null;
+  }
+  if (proxyFromEnv.indexOf('://') === -1) {
+    proxyFromEnv = 'http' + '://' + proxyFromEnv;
```

Comment:
since in this scenario customer specified 'my.pro.xy' and we using something else instead, http://my.pro.xy, can we please log a INFO that protocol was missing from the environment proxy, and `http` is automatically assumed?

Reply:
I added it.